### PR TITLE
Added prepareUpload to NotificationClient.

### PR DESCRIPTION
**Pull request by servingUpAces**
This method prepares the contents of a file to be uploaded, the result of the method is added to the personalisation of a sendEmail.
The user will see a link in the email, when clicked the user can download the document to their device.

## Checklist

- [x] I’ve used the pull request template
- [x] I’ve written unit tests for these changes
- [x] I’ve update the documentation (in `DOCUMENTATION.md`)
- [x] I’ve bumped the version number (in `src/main/resources/application.properties`)
      and run the `update_version.sh` script

### CODE REVIEW

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -231,6 +231,40 @@ If the request is not successful, the client returns a `NotificationClientExcept
 |`429`|`[{`<br>`"error": "TooManyRequestsError",`<br>`"message": "Exceeded send limits (LIMIT NUMBER) for today"`<br>`}]`|Refer to [service limits](#service-limits) for the limit number|
 |`500`|`[{`<br>`"error": "Exception",`<br>`"message": "Internal server error"`<br>`}]`|Notify was unable to process the request, resend your notification|
 
+##Send a document by email
```
Comment (jonathanglassman) [inline DOCUMENTATION.md]: Add a space between ## and Send - sometimes we get an error in the tech doc tool when there aren't spaces between the # and the title

File: DOCUMENTATION.md [hunk 2]
```diff
@@ -231,6 +231,40 @@ If the request is not successful, the client returns a `NotificationClientExcept
 |`429`|`[{`<br>`"error": "TooManyRequestsError",`<br>`"message": "Exceeded send limits (LIMIT NUMBER) for today"`<br>`}]`|Refer to [service limits](#service-limits) for the limit number|
 |`500`|`[{`<br>`"error": "Exception",`<br>`"message": "Internal server error"`<br>`}]`|Notify was unable to process the request, resend your notification|
```
Comment (allait) [inline DOCUMENTATION.md]: There are additional errors in the python docs related to document uploads (like "unsupported document type" and "document failed virus scan") that should probably be added to this list

Comment (servingUpAces) [reply DOCUMENTATION.md]: Good point - done

File: src/main/java/uk/gov/service/notify/NotificationClient.java [hunk 3]
```diff
@@ -285,6 +287,26 @@ public ReceivedTextMessageList getReceivedTextMessages(String olderThanId) throw
...
+     * Use the prepareUpload method when uploading a document via sendEmail.
+     * The prepareUpload method creates a <code>JSONObject</code> which will need to be added to the personalisation map.
+     *
+     * @param documentContents byte[] of the document
+     * @return <code>JSONObject</code> a json object to be added to the personalisation is returned
+     * @throws UnsupportedEncodingException exception thrown if unable to create a String using "IS0-8859-1" character set.
+     */
+    public static JSONObject prepareUpload(final byte[] documentContents) throws UnsupportedEncodingException, NotificationClientException {
+        if (documentContents.length > 2*1024*1024){
+            throw new NotificationClientException("File size must be under 2MB");
```
Comment (allait) [inline src/main/java/uk/gov/service/notify/NotificationClient.java]: @karlchillmaid suggested `Document is larger than 2MB` for the error message

**Pull request merged**